pkg/R/RDD.R

```diff
@@ -1348,16 +1363,16 @@ setMethod("unionRDD",
           function(x, y) {
             if (x@env$serialized == y@env$serialized) {
               jrdd <- callJMethod(getJRDD(x), "union", getJRDD(y))
-              union.rdd <- RDD(jrdd, x@env$serialized)
+              union.rdd <- RDD(jrdd, x@env$serialized, colNames = x@colNames)
             } else {
               # One of the RDDs is not serialized, we need to serialize it first.
-              if (!x@env$serialized) {
+              if (x@env$serialized == "string") {
```

Comment:
Same here, to run the union correctly I think we need to do `!= byte` ? It'll be good to add a testcase which takes 2 data frames, calls `toRDD` on them and then calls `union` on the results to see if this works.